clang/test/Sema/attr-counted-by-or-null-last-field.c

```diff
@@ -80,10 +81,17 @@ struct found_outside_of_struct {
   struct bar ** ptr __counted_by_or_null(global); // expected-error {{field 'global' in 'counted_by_or_null' not inside structure}}
 };
 
+#ifndef LATE_PARSING_ENABLED
 struct self_referrential {
   int bork;
   struct bar *self[] __counted_by_or_null(self); // expected-error {{use of undeclared identifier 'self'}}
 };
+#else
+struct self_referrential {
+  int bork;
+  struct bar *self[] __counted_by_or_null(self); // expected-error {{'counted_by_or_null' only applies to pointers; did you mean to use 'counted_by'?}}
+};
+#endif
```

Comment:
Rather than duplicate the test, please use `-verify=whatever` to use different prefixes for the `expected-error` comments.

Reply:
Good idea. Let me fix that.